### ISSUE TITLE
Improve release workflow

### ISSUE THREAD

**Issue by wch** [posit-dev/py-shiny#542]
Currently, to make a release to PyPI, we:

- Tag a commit and push
- Make a GitHub release

The GitHub Action workflow does the rest This is pretty simple, but there is a problem: if the release fails for some reason, like if it has trouble uploading to PyPI, or if there's some bug in the PyPI publishing script (which can happen because it's not used very often) then the "release" is not actually a release, and we need to go back, delete the GitHub Release and the tag.

It would be nice if we could push a branch with a certain prefix, like `release-v1.0.0`, and the action would attempt publishing to PyPI, and then if successful it would tag it and create a GitHub Release. It could also delete the branch.

**Comment by wch** [comment]
Yes, we'd definitely want it for other repos with Python packages.

**Comment by nealrichardson** [comment]
To include: https://github.com/rstudio/py-shiny/issues/508
